arch/xtensa/src/common/xtensa_swi.h

```diff
@@ -0,0 +1,68 @@
+/****************************************************************************
+ * arch/xtensa/src/common/xtensa_swi.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __ARCH_XTENSA_SRC_COMMON_XTENSA_SWI_H
+#define __ARCH_XTENSA_SRC_COMMON_XTENSA_SWI_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <arch/xtensa/core.h>
+#include <arch/xtensa/xtensa_corebits.h>
+
+/* Select software interrupt number for context-switch.
+ * The SW interrupt level must larger than XCHAL_SYSCALL_LEVEL
+ * and not larger than XCHAL_EXCM_LEVEL.
+ * For that it can generate interupt when up_irq_save.
+ * and not generate interupt when up_irq_disable.
+ * Otherwise return error if no one interrupt find.
+ */
```

Comment:
```suggestion
/* Select the software interrupt number for context-switch.
 * The SW interrupt level must be greater than XCHAL_SYSCALL_LEVEL
 * and less than XCHAL_EXCM_LEVEL.
 * So that we can generate an interrupt when up_irq_save is called.
 * and not generate interrupt when up_irq_disable is called.
 * Return an error if no suitable software interrupt was found.
 */
```